platform_plugin_elm_credentials/api/credential_builder.py

```diff
@@ -0,0 +1,136 @@
+"""Builds ELM credentials."""
+from django.conf import settings
+
+from platform_plugin_elm_credentials.api.serializers import (
+    Address,
+    AwardedBy,
+    AwardingBody,
+    CountryCode,
+    CredentialSubject,
+    DeliveryDetails,
+    DisplayParameter,
+    HasClaim,
+    Issuer,
+    Location,
+    PrimaryLanguage,
+)
+from platform_plugin_elm_credentials.api.utils import get_fullname, to_iso_format
+
+
+class CredentialBuilder:
+    """
+    Builds ELM credentials for a user based on a given course and certificate.
+
+    This class provides a convenient way to construct ELM credentials for a user who has
+    successfully completed a specified course and received a certificate of completion.
+
+    Attributes:
+        course (CourseOverview): The course for which credentials are being built.
+        user (User): The user for whom credentials are being built.
+        certificate (GeneratedCertificate): The certificate associated with the user's completion of the course.
+        additional_params (dict): Additional parameters for the credential.
+
+    Properties:
+        full_name (str): The full name of the user.
+        primary_language (PrimaryLanguage): The primary language of the credential.
+        country_code (CountryCode): The country code of the credential.
+
+    Methods:
+        build() -> dict:
+            Constructs and returns the ELM credential in the form of a dictionary.
+    """
+
+    def __init__(self, course, user, certificate, additional_params):
+        self.course = course
+        self.user = user
+        self.certificate = certificate
+        self.additional_params = additional_params
+
+    @property
+    def full_name(self) -> str:
+        """
+        Get the full name of the user.
+
+        Returns:
+            str: The full name of the user.
+        """
+        return self.user.profile.name
+
+    @property
+    def primary_language(self) -> PrimaryLanguage:
+        """
+        Get the primary language of the credential.
+
+        Returns:
+            PrimaryLanguage: The primary language of the credential.
+        """
+        primary_language_id = getattr(
+            settings, "ELM_CREDENTIALS_PRIMARY_LANGUAGE_ID", None
+        )
+        if primary_language_id:
+            return PrimaryLanguage(id=primary_language_id)
+        return PrimaryLanguage()
```

Comment:
The configured language is something you can get from the existing Django setting. I think you can use the `LANGUAGE_CODE`

Reply:
Done